src/Microsoft.AspNetCore.Mvc.TagHelpers/CacheTagHelper.cs

```diff
@@ -108,18 +108,24 @@ public CacheTagHelper(IMemoryCache memoryCache, HtmlEncoder htmlEncoder) : base(
 
                                 content = await result;
                             }
+                            
+                            tcs.SetResult(content);
```

Comment:
Looks like setting this might be short-circuiting an extra (unnecessary loop). Consider this:

1. Caller A gets a cache miss and enters the critical sections sets a TCS.Task as the Cache Entry
2. Caller B does a TryGetValue and gets the TCS.Task
3. Caller A replaces the CacheEntry.Value with the task returned from the TagHelper.
4. Caller A does TCS.SetResult(null);
5. Caller A's output is valid html. Caller B's output is null. It has to redo the `TryGetValue` at step 2.

It's pretty unrelated to the actual change, but might help us remove the `while` loop.